### PR TITLE
RxSwift integration

### PR DESCRIPTION
**Pull request by truizlop**
This PR adds integration with RxSwift based on the typeclasses in Effects. Addresses issue #43. The included classes are:

- `SingleK<A>`: wrapper over `Single<A>`
- `MaybeK<A>`: wrapper over `Maybe<A>`
- `ObservableK<A>`: wrapper over `Observable<A>`

`Flowable` is not present in RxSwift; therefore, it has not been ported from Arrow.

### CODE REVIEW

File: Tests/BowTests/Effects/IOTest.swift [hunk 1]
```diff
@@ -58,6 +58,6 @@ class IOTest: XCTestCase {
     }
     
     func testAsyncContextLaws() {
-        AsyncContextLaws<ForIO>.check(asyncContext: IO<Int>.asyncContext(), monadError: IO<Int>.monadError(), eq: self.eq, gen : { CategoryError.arbitrary.generate })
+        AsyncLaws<ForIO>.check(async: IO<Int>.asyncContext(), monadError: IO<Int>.monadError(), eq: self.eq, gen : { CategoryError.arbitrary.generate })
```
Comment (pakoito) [inline Tests/BowTests/Effects/IOTest.swift]: Rename IO async here :)

File: Tests/BowTests/Effects/Rx/MaybeKTest.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        MonadErrorLaws.check(monadError: MaybeK<Int>.monadError(), eq: eq, gen: constant(CategoryError.unknown))
+    }
+    
+    func testApplicativeErrorLaws() {
+        ApplicativeErrorLaws.check(applicativeError: MaybeK<Int>.applicativeError(), eq: eq, eqEither: MaybeKEitherEq(), gen: constant(CategoryError.unknown))
+    }
+    
+    func testAsyncLaws() {
+        AsyncLaws.check(async: MaybeK<Int>.async(), monadError: MaybeK<Int>.monadError(), eq: eq, gen: constant(CategoryError.unknown))
+        AsyncLaws.check(async: MaybeK<Int>.effect(), monadError: MaybeK<Int>.monadError(), eq: eq, gen: constant(CategoryError.unknown))
```
Comment (pakoito) [inline Tests/BowTests/Effects/Rx/MaybeKTest.swift]: Effect inherits from Async, why do we need both?

File: Tests/BowTests/Effects/IOTest.swift (hunk 1 above)
Comment (truizlop) [reply Tests/BowTests/Effects/IOTest.swift]: 👍🏼

File: Tests/BowTests/Effects/Rx/MaybeKTest.swift (hunk 2 above)
Comment (truizlop) [reply Tests/BowTests/Effects/Rx/MaybeKTest.swift]: You are right, no need.

**Pull request merged**